src/spatial/models/report_datafetcher.py

```
@@ -0,0 +1,281 @@
+
+    def generate_report_with_gemini(self, audience):
+        prompt = self._generate_prompt(audience)
+        try:
+            response = self.gemini_model.generate_content(prompt)
+            gemini_output = response.text
+            return self._prepare_report_json(gemini_output)
+        except Exception as e:
+            print(f"Error: {e}")
+            return None
+    # Generate report with customised prompt
+    @lru_cache(maxsize=64)  # Cache up to 64 most recent reports
```

Ruff (0.8.0)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)